stacks_queues/queue.py

```diff
@@ -15,47 +18,71 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = self.rear = 0
+
+        if self.full():
+            # raise QueueFullException("Queue is full")
+            buffer_increase = int(self.buffer_size * 0.5)
+            self.store = self.store[:self.rear] + [None] * buffer_increase + self.store[self.front:]
+            self.buffer_size += buffer_increase
+            self.front += buffer_increase
```

Comment:
Nicely done. Consider moving this logic to a helper, such as `grow_buffer`. Also, consider adding some comments here. For instance, I was initially unsure if this would properly handle conditions both where the front is ahead of the rear, and the reverse. But the way you're counting the where the front and rear are guarantees that when the buffer is full, the front will equal the rear. So line 34 is really taking anything up to the buffer split point, injecting a new chunk of space, then taking the remainder. The use of both front and rear makes it seem like there is a meaningful difference here.